source/validate_cfg.cpp

```diff
@@ -268,21 +271,52 @@ void UpdateContinueConstructExitBlocks(
   }
 }
 
+tuple<string, string, string> ConstructNames(ConstructType type) {
+  string construct_name, header_name, exit_name;
+
+  switch (type) {
+    case ConstructType::kSelection:
+      construct_name = "selection";
+      header_name = "selection header";
+      exit_name = "merge block";
+      break;
+    case ConstructType::kLoop:
+      construct_name = "loop";
+      header_name = "loop header";
+      exit_name = "merge block";
+      break;
+    case ConstructType::kContinue:
+      construct_name = "continue";
+      header_name = "continue target";
+      exit_name = "back-edge block";
+      break;
+    case ConstructType::kCase:
+      construct_name = "case";
+      header_name = "case block";
+      exit_name = "exit block";  // TODO(umar): there has to be a better name
+      break;
+    default:
+      assert(1 == 0 && "Not defined type");
+  }
+
+  return make_tuple(construct_name, header_name, exit_name);
+}
+
 /// Constructs an error message for construct validation errors
 string ConstructErrorString(const Construct& construct,
                             const string& header_string,
                             const string& exit_string,
                             bool post_dominate = false) {
-  string construct_name;
-  string header_name;
-  string exit_name;
-  string dominate_text;
+  string construct_name, header_name, exit_name, dominate_text;
   if (post_dominate) {
     dominate_text = "is not post dominated by";
   } else {
     dominate_text = "does not dominate";
   }
 
+  tie(construct_name, header_name, exit_name) =
+      ConstructNames(construct.type());
+
   switch (construct.type()) {
```

Comment:
This should be dead code since you've replaced it with the call to ConstructNames